src/pages/index.js

```diff
@@ -1,256 +1,229 @@
-    card.domDeleteCard();
-    deleteConfirmModal.close();
-  });
-}
+// PREVIEW IMAGE
+const imagePopup = new PopupWithImage("#preview-image-modal");
 
-// AVATAR
-function handleAvatarChangeSubmit(Url) {
-  console.log(Url.avatarUrl);
-  api.updateAvatar(Url.avatarUrl).then((res) => {
-    console.log(res);
-  });
-  user.setAvatarPic(Url.avatarUrl);
-  submitButton.textContent = "Saving";
-  avatarChangeModal.close();
-}
+/////////////////////////////////////////////////////////   FUNCTION   ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// PREVIEW IMAGE
+function handleImageClick(name, link) {
+  imagePopup.open({ name, link });
+}
 
-// LIKE & UNLIKE FUNCTION
-function handleCardLike(cardData) {
-  if (cardData.isLiked === false) {
-    api.addLikes(cardData._id).then((res) => {
-      console.log(res);
-    });
-  } else {
-    api.removeLikes(cardData._id).then((res) => {
-      console.log(res);
+// LIKE
+function handleLikeClick(card) {
+  api
+    .setCardLikes(card.id, card.isLiked)
+    .then((newCardData) => {
+      card.updateLikes(newCardData.isLiked);
+    })
+    .catch((err) => {
+      console.log("Failed to update card likes status:", err);
     });
-    console.log(cardData._id);
-  }
 }
 
-//////////////////////////////////////////////////  ADD EVENT LISTENERS ///////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////   EVENT LISTENERS  ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-// PROFILE
+// PROFILE EDIT
 profileEditButton.addEventListener("click", () => {
-  const userInput = user.getUserInfo();
-  editFormValidator.resetValidation();
-  profileNameInput.value = userInput.name;
-  profileDescriptionInput.value = userInput.description;
-  submitButton.textContent = "Save";
-  editProfileModal.open();
+  const currentUserInfo = userInfo.getUserInfo();
+  profileTitleInput.value = currentUserInfo.name;
+  profileDescriptionInput.value = currentUserInfo.about;
+  editProfileValidator.resetValidation();
+  profileEditPopup.open();
 });
 
-// CARD
-cardAddButton.addEventListener("click", () => {
-  submitButton.textContent = "Save";
-  newCardModal.open();
-  cardFormValidator.toggleButtonState();
+addCardPopup.setEventListeners();
+
+profileAddButton.addEventListener("click", () => {
+  addCardValidator.resetValidation();
+  addCardPopup.open();
 });
 
-// AVATAR
-avatarPicture.addEventListener("click", () => {
-  submitButton.textContent = "Save";
-  avatarChangeModal.open();
-  avatarFormValidator.toggleButtonState();
+// CHANGE AVATAR
+changeAvatarImageButton.addEventListener("click", () => {
+  avatarFormValidator._toggleButtonState();
```

Comment:
`_toggleButtonState`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.